### DISCUSSION TITLE
Big breaking change update to workflows probably coming next weekend

### DISCUSSION THREAD

**Discussion by SomeOddCodeGuy** [SomeOddCodeGuy/WilmerAI#10] [General]
There's probably like 4 of us worldwide that use this, but just in case this might affect anyone I want to throw this out there:

tl;dr- Next weekend I'll be breaking all the existing workflows, so back your stuff up! Newer, fancier, hopefully improved workflow prompts should be incoming for everything, as well as much easier to manage presets and endpoints.

------------------

I've been working pretty hard the past week or two on finally fixing up the prompts, fixing up the presets, and adding proper KoboldCpp support. What exists currently in the workflows and presets were performing... ok, but not to the level I wanted, and really of those are the result of my first swing at trying to make things work. Now that I've started to have more time to really stress test the quality, I've realized I wanted better than what Wilmer was giving me.

Next weekend's update will be a breaking change for all of the existing workflows. I'm upending a lot of my original theories about how to prompt, including getting rid of all second person prompts and all references to "roleplay". For the latter- don't fret if you use Wilmer for that; this change won't break it at all. In fact it might do better at that, I'm betting. 

The existence of that term in the prompts was most due to the prompts being in second person, and also because of my assumption that all personas are a form of Roleplay (ie- JARVIS in Ironman would be a roleplay. The LLM has no persona; it's 'pretending' to be an entity named JARVIS). There have been quite a few articles/papers talking about the effect of personas on LLMs, and I was leaning into that.

Some conversations lead me to question, however, the efficacy of spelling that out in the prompt and also how well second person works in general for prompting. So now I'm shifting everything to third person. _"You are an expert AI in a roleplay with a user in an online conversation. You can find the instructions for the roleplay here:\n[\n{chat_system_prompt}\n]\n"_ would be replaced with something like _"A user has sent a new message in an online conversation with an AI. The instructions for that conversation, as well as any personas involved, can be found... ... please continue the conversation, portraying the persona that has been directed by the system instructions, and..."_ or something along those lines. 

These changes accomplish the same thing, but takes a different route. And so far, I'm really liking the results.

Additionally, I've leaned more into injecting context into the system prompt where possible, rather than trying to utilize a user prompt. I've started to see slightly better results in some places, but I think I need more testing to be sure on that. Still, I think it would be better to move Wilmer in this direction for now.

Together, I've had a lot better results (at least it feels that way to me). I've been toying with these prompts for a few weeks now, and feel pretty confident that I want to push them out.

Additionally, I'm changing how some of the endpoints and presets are named, to make it easier for new users and for me (mostly me lol. I realized changing a model at an endpoint was getting really old, really fast. And I change models a LOT to test).

So anyhow, wanted to give a heads up in case anyone happened to see. [Recently I added the ability to move your configs to a new location and refence them on startup](https://github.com/SomeOddCodeGuy/WilmerAI/pull/8), so maybe take a peek at that.
